### PR TITLE
feature: sync design tokens only

### PR DESCRIPTION
**Pull request by andrico-anima**
https://www.loom.com/share/0fd7aee4ddaa40b08468c18e567d75fa

https://app.asana.com/0/1204315207094721/1204485586761684/f

### CODE REVIEW

File: src/helpers/file-system.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import fs from 'fs-extra';
+
+export const getFileIfExists = async <T>(path?: string): Promise<T | undefined> => {
```
Comment (Datlyfe) [inline src/helpers/file-system.ts:3]:
```suggestion
export const getFileOrThrow= async <T>(path?: string): Promise<T | undefined> => {
```

File: src/api/storybook.ts [hunk 2]
```diff
@@ -160,6 +166,64 @@ export const getOrCreateStorybook = async (
   };
 };
 
+export const getOrCreateStorybookForDesignTokens = async (
+  token: string,
+  raw_ds_tokens: Record<string, unknown> = {},
+): Promise<getOrCreateStorybookResponse> => {
+  const transaction = getCurrentHub().getScope()?.getTransaction();
+  const spanGetOrCreate = transaction?.startChild({
+    op: 'get-or-create-storybook',
```
Comment (Datlyfe) [inline src/api/storybook.ts:175]:
```suggestion
    op: 'get-or-create-storybook-for-design-tokens',
```

File: src/commands/sync-design-tokens.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    loader.stop();
+    log.yellow(
+      `Storybook token not found. Please provide a token using the --token flag or the STORYBOOK_ANIMA_TOKEN environment variable.`,
+    );
+    await exitProcess();
+  }
+
+  const authSpan = transaction.startChild({ op: 'authenticate' });
+
+  const response = await authenticate(token);
```
Comment (Datlyfe) [inline src/commands/sync-design-tokens.ts:97]: There is a small issue here, authenticating with the token will set `is_storybook_paired` to `true` on the api, which is a side effect that we need to remove and do it explicity not everytime the token is being verified, something to keep in mind but lets create a ticket about this please.

File: src/commands/sync-design-tokens.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+  log.green(`  - ${stage} ...OK`);
+
+  const storybook = await getOrCreateStorybookForDesignTokens(
+    token,
+    designTokens,
+  );
+
+  const spanUpload = transaction.startChild({
+    op: 'upload-process',
```
Comment (Datlyfe) [inline src/commands/sync-design-tokens.ts:132]:
```suggestion
    op: 'update-design-tokens',
```

**Review by Datlyfe — CHANGES_REQUESTED** [review]
Great job, couple comments

File: src/helpers/file-system.ts (hunk 1 above)
Comment (andrico-anima) [reply src/helpers/file-system.ts:3]: done!

File: src/api/storybook.ts (hunk 2 above)
Comment (andrico-anima) [reply src/api/storybook.ts:175]: done!

File: src/commands/sync-design-tokens.ts (hunk 3 above)
Comment (andrico-anima) [reply src/commands/sync-design-tokens.ts:97]: Will do!

Comment (andrico-anima) [reply src/commands/sync-design-tokens.ts:97]: https://app.asana.com/0/1203335192450540/1204521282059336/f

File: src/commands/sync-design-tokens.ts (hunk 4 above)
Comment (andrico-anima) [reply src/commands/sync-design-tokens.ts]: done!

**Comment by andrico-anima** [comment]
Thanks for reviewing @Datlyfe

**Review by Datlyfe — APPROVED** [review]

**Pull request merged**